quail/processes/wps_climdex_sdi.py

```diff
@@ -0,0 +1,146 @@
+import os
+from rpy2 import robjects
+from pywps import Process, LiteralInput
+from pywps.app.Common import Metadata
+from pywps.app.exceptions import ProcessError
+
+from wps_tools.utils import log_handler, collect_args, common_status_percentages
+from wps_tools.io import log_level
+from quail.utils import get_package, logger, load_rdata_to_python, save_python_to_rdata
+from quail.io import climdex_input, ci_name, output_file, rda_output, vector_name
+
+
+class ClimdexSDI(Process):
+    """
+    Cold or warm spell duration index
+
+    The warm spell duration index is defined as the number
+    of days each year which are part of a "warmspell". A
+    "warm spell" is defined as a sequence of 6 or more days
+    in which the daily maximumtemperature exceeds the 90th
+    percentile of daily maximum temperature for a 5-day
+    running windowsurrounding this day during the baseline
+    period.
+    """
+
+    def __init__(self):
+        self.status_percentage_steps = dict(
+            common_status_percentages,
+            **{
+                "load_rdata": 10,
+                "save_rdata": 90,
+            },
+        )
+        inputs = [
+            climdex_input,
+            ci_name,
+            output_file,
+            LiteralInput(
+                "func",
+                "Function to compute",
+                abstract="Compute climdex.wsdi (Warm spell duration index)",
+                allowed_values=["wsdi"],
```

Comment:
I guess another `sdi` will be added here?

Reply:
csdi, cold spell duration index!